src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -1164,37 +1175,30 @@ private unsafe SocketError FinishOperationConnect()
         private void CompleteCore()
         {
             _strongThisRef.Value = null; // null out this reference from the overlapped so this isn't kept alive artificially
-            if (_singleBufferHandleState != SingleBufferHandleState.None)
+
```

Comment:
You decided that making CompleteCore inlineable wasn't worthwhile?

Reply:
No, I just haven't fixed it yet